### PR TITLE
[TASK] Introduce custom confval directive for basic types

### PR DESCRIPTION
**Pull request by linawolf**
Removed all types that have not been referenced anymore

references #658
releases: main, 11.5

### CODE REVIEW

**Review by tomwarwick — APPROVED** [review]

**Review by brotkrueml — COMMENTED** [review]
It is wanted that the "wrap" link is displayed in another way than the other links?
image
It looks kind of wrong for me.
Example: [Documentation/ContentObjects/Content/Index.rst](https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-Typoscript/pull/669/files#diff-bfd7f1707ffe067b3dd55952b91d8facac4982abdb073c0c29d0692f979d08ed)

**Comment by linawolf** [comment]
Wrap is a simple type, the others are functions. In this ticket I only solved the simple types. The functions will follow in follow ups

File: Documentation/DataTypes/Index.rst [hunk 1]
```diff
@@ -22,782 +22,400 @@ examples.
...
-      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
-
-      page.10.settings.someTime = %T
-
-
-.. index:: Simple data types; string
-.. _data-type-string:
-.. _data-type-str:
-.. _data-type-value:
+..  t3-data-type:: strftime-conf
```
Comment (brotkrueml) [inline Documentation/DataTypes/Index.rst:369]: I suggest to keep the TypoScript examples.

Comment (linawolf) [reply Documentation/DataTypes/Index.rst:369]: The imageResource was not linked anymore, this is just a list of types. How to use the GIFBUILDER is described on the GIFBUILERs page

Comment (brotkrueml) [reply Documentation/DataTypes/Index.rst:369]: It's about strftime, not GIFBUILDER. All the examples are gone, like:

    page.10.settings.myDate = %e:%m:%y

**Review by brotkrueml — APPROVED** [review]

File: Documentation/DataTypes/Index.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/DataTypes/Index.rst:369]: I don't think we should copy PHP documentation into TYPO3 docs

**Pull request merged**